### PR TITLE
Bugfix #517

### PR DESCRIPTION
**Pull request by PauloAmora**
This PR solves Bug #517

The bug was caused by the order of operations in `SimpleOptimizer`, since `ProjectPlan` was always placed below `OrderByPlan` in the plan tree, the `OrderByPlan` would order only the output.

The solution was to check if there was an `order_by` before creating the `ProjectPlan`, and take care to not recreate it in a later point. This required moving the logic of `order_by` and `order_by limit` to separate functions.

### CODE REVIEW

**Comment by Michael-Tieying-Zhang** [comment]
@malin1993ml Lin, the code (`ProjectPlan`) @PauloAmora modified is the stuff in the original simple_optimizer. Could you review the code?

File: src/optimizer/simple_optimizer.cpp [hunk 1]
```diff
@@ -1508,5 +1363,164 @@ bool SimpleOptimizer::UnderlyingSameOrder(planner::AbstractPlan* select_plan,
...
+        "Underlying plan has the same ordering output with"
+        "order_by plan without limit");
+    order_by_plan->SetUnderlyingOrder(true);
+  }
+  return std::move(order_by_plan);
+}
+
+std::unique_ptr<planner::AbstractPlan> SimpleOptimizer::CreateOrderByLimitPlan(
+    parser::SelectStatement* select_stmt, planner::AbstractPlan* child_plan,
+    catalog::Schema* schema, std::vector<oid_t> column_ids, bool is_star) {
```
Comment (malin1993ml) [inline src/optimizer/simple_optimizer.cpp]: This function has lots of code in common with CreateOrderByPlan(). Can you remove the duplicated code, reuse CreateOrderByPlan() here to generate the order_by_plan, and then set the limit flags and add the limit plan?

Comment (PauloAmora) [inline src/optimizer/simple_optimizer.cpp]: Got it, I tried not to mess around with the already present code, but I'll do that.

**Comment by PauloAmora** [comment]
Sorry about that, I messed up the rebase (forgot to extract my commits as patches). I'll submit another PR with the revision fixes.

**Pull request closed without merging**